lib/services/services.c

```diff
@@ -221,8 +221,7 @@ resources_action_create(const char *name, const char *standard,
     }
 
     if (is_set(ra_caps, pcmk_ra_cap_params)) {
-        op->params = params;
-        params = NULL; // so we don't free them in this function
+        op->params = crm_str_table_dup(params);
```

Comment:
While your approach makes more sense, resources_action_create() is part of the public API, so we should avoid backward-incompatible changes (i.e. a hypothetical existing caller would get a memory leak from this). Let's flip this and make crm_resource duplicate the table before sending it to resources_action_create().